### ISSUE TITLE
Expand from `public`/`private` to `public`/`private`/`local`

### ISSUE THREAD

**Issue by mikewest** [WICG/cors-rfc1918#3]
As opposed to this document's `private`/`public` distinction, Opera Presto <a href="https://code.google.com/p/chromium/issues/detail?id=378566#c6">apparently</a> had three levels: `internet`, `intranet`, and `localhost`. The same restrictions were applied from internet to intranet as from intranet to localhost. This document should do the same.

**Comment by Ajedi32** [comment]
"Private" is actually pretty difficult to define for IPv6 addresses, where there's not necessarily a separate address space for "internal" networks. [RFC3879](https://tools.ietf.org/html/rfc3879) touches on this point a bit, particularly [section 2.5](https://tools.ietf.org/html/rfc3879#section-2.5).

**Comment by annevk** [comment]
It seems that https://tools.ietf.org/html/rfc4193 might provide RFC1918 for IPv6?

**Comment by mikewest** [comment]
My understanding (that is, barely remembered conversations with @sleevi, @bslassey, and @davidben) is that there are some delineations we can rely upon in IPv6, but that things are more fluidly deployed.

My plan for shipping this in Chrome would be to have a default mode that defined `intranet` in terms of the user's locally-defined network prefix, and a set of configuration options for administrators to define more granular ranges. My hope is that the prefix-based approach would exclude the printers and routers on my local network (via their local address), which is the main risk for home users, and responsible sysadmins could take responsibility for the rest. :)

**Comment by hsivonen** [comment]
> My plan for shipping this in Chrome would be to have a default mode that defined `intranet` in terms of the user's locally-defined network prefix

Did this ever end up prototyped? How did it go?

**Comment by mikewest** [comment]
I prototyped it in Blink. It didn't work because I'm dumb and DNS all happens way outside of Blink. Then ~4 years passed, and in the meantime CORS has been refactored to be in a place where it might actually be possible to implement this after socket connection.

https://groups.google.com/a/chromium.org/d/msg/blink-dev/EeGg7TxW6U4/9aohper3BQAJ is a first step towards that world, but I dropped it on the floor, and it's only now being picked up again (by @camillelamy). Q3! Probably!
